plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMigrateProcedure.java

```diff
@@ -226,18 +284,98 @@ public void testMigrateUnsupportedColumnType()
+
+        assertUpdate(session, "DROP TABLE " + icebergTableName);
+    }
+
+    @Test
+    public void testMigrateColumnTypeTimestampMillisOnFormatAvro()
+    {
+        String tableName = "test_migrate_timestamp_millis_avro" + randomNameSuffix();
+        String hiveTableName = "hive.tpch." + tableName;
+        String icebergTableName = "iceberg.tpch." + tableName;
+        assertUpdate("CREATE TABLE " + hiveTableName + "  WITH (format='" + IcebergFileFormat.AVRO + "') AS SELECT timestamp '2021-01-01 00:00:00.000' x", 1);
         assertQuery("SELECT * FROM " + hiveTableName, "VALUES timestamp '2021-01-01 00:00:00.000'");
         assertQueryFails("SELECT * FROM " + icebergTableName, "Not an Iceberg table: .*");
 
+        assertQueryFails("CALL iceberg.system.migrate('tpch', '" + tableName + "')", "Timestamp precision \\(3\\) not supported for Iceberg with Avro file format. Use \"timestamp\\(6\\)\" instead.");
+
         assertUpdate("DROP TABLE " + hiveTableName);
     }
 
+    @DataProvider
+    public static Object[][] timestampPrecisionsAndValuesParquet()
+    {
+        return new Object[][] {
+                {HiveTimestampPrecision.MILLISECONDS, "2021-01-01 06:12:12.123", "2021-01-01 12:12:12.123"},
+                {HiveTimestampPrecision.MICROSECONDS, "2021-01-01 06:12:12.123456", "2021-01-01 12:12:12.123456"}
+        };
+    }
+
+    @Test(dataProvider = "timestampPrecisionsAndValuesParquet")
+    public void testMigrateColumnTypeTimestampOnFormatParquet(HiveTimestampPrecision precision, String given, String expected)
+    {
+        String tableName = "test_migrate_timestamp_parquet" + randomNameSuffix();
+        String hiveTableName = "hive.tpch." + tableName;
+        String icebergTableName = "iceberg.tpch." + tableName;
+
+        Session session = timestampPrecisionSession(precision);
+
+        assertUpdate(session, "CREATE TABLE " + hiveTableName + " (x timestamp(" + precision.getPrecision() + ")) WITH (format='" + IcebergFileFormat.PARQUET + "') ");
+        assertUpdate(session, "INSERT INTO " + hiveTableName + " (x) values timestamp '" + given + "'", 1);
+        assertQuery(session, "SELECT * FROM " + hiveTableName, "VALUES timestamp '" + given + "'");
+        assertQueryFails(session, "SELECT * FROM " + icebergTableName, "Not an Iceberg table: .*");
+
+        assertQuerySucceeds(session, "CALL iceberg.system.migrate('tpch', '" + tableName + "')");
+
+        assertQuery(session, "SELECT * FROM " + icebergTableName, "VALUES timestamp '" + expected + "'");
+
+        assertUpdate(session, "DROP TABLE " + icebergTableName);
+    }
+
+    private Session timestampPrecisionSession(HiveTimestampPrecision hiveTimestampPrecision)
+    {
+        return Session.builder(getSession())
+                .setCatalogSessionProperty("hive", "timestamp_precision", hiveTimestampPrecision.name())
+                .build();
+    }
+
```

Comment:
Could you add a new test to `TestIcebergProcedureCalls` for verifying that Spark can read migrated tables from timestamp(3) type?

Reply:
I added test for ORC file format.
For Parquet I have stumbled on some issues that I'm trying to resolve (wip)

Reply:
Btw which product test is the most suitable to run the test TestIcebergProcedureCalls? Is the  singlenode-spark-iceberg will be sufficient?